hedera-node/src/main/java/com/hedera/services/state/expiry/renewal/RenewalHelper.java

```diff
@@ -116,20 +124,23 @@ private List<MapAccessType> workFor(final MerkleAccount payer, final MerkleAccou
     void renewWith(long fee, long renewalPeriod) {
         assertPayerAccountForRenewalCanAfford(fee);
 
-        final var mutableAccount = lookup.getMutableAccount(classifier.getLastClassifiedNum());
-        final long newExpiry = mutableAccount.getExpiry() + renewalPeriod;
-        mutableAccount.setExpiry(newExpiry);
+        final var lastClassifiedAccount = classifier.getLastClassifiedNum().toGrpcAccountId();
+        final var payerForLastClassified =
+                classifier.getPayerNumForLastClassified().toGrpcAccountId();
 
-        final var mutablePayerForRenew =
-                lookup.getMutableAccount(classifier.getPayerNumForLastClassified());
-        final long newBalance = mutablePayerForRenew.getBalance() - fee;
-        mutablePayerForRenew.setBalanceUnchecked(newBalance);
+        accountsLedger.begin();
+        final long newExpiry =
+                ((long) accountsLedger.get(lastClassifiedAccount, EXPIRY)) + renewalPeriod;
+        accountsLedger.set(lastClassifiedAccount, EXPIRY, newExpiry);
 
-        final var fundingAccount = dynamicProperties.fundingAccount();
-        final var fundingId = fromAccountId(fundingAccount);
-        final var mutableFundingAccount = lookup.getMutableAccount(fundingId);
-        final long newFundingBalance = mutableFundingAccount.getBalance() + fee;
-        mutableFundingAccount.setBalanceUnchecked(newFundingBalance);
+        nonHapiFeeCharging.chargeNonHapiFee(
+                EntityId.fromGrpcAccountId(payerForLastClassified),
+                lastClassifiedAccount,
+                fee,
+                accountsLedger,
+                INSUFFICIENT_PAYER_BALANCE);
+
+        accountsLedger.commit();
```

Comment:
⚠️ Since this `commit()` could trigger staking rewards, we must refactor to ensure `ExpiryRecordsHelper` uses the singleton `SideEffectsTracker` to set the hbar `transferList` (and `staking_rewards_paid`).

Reply:
Fixed !